src/model/jwks.rs

```diff
@@ -106,20 +107,28 @@ pub struct Jwk {
     alg: String,
     kid: String,
     r#use: String,
+    x5c: Vec<String>,
     #[serde(skip_serializing)]
     private_key_pem: Vec<u8>,
 }
 
 impl Jwk {
-    pub fn new(rsa: Option<Rsa<Private>>) -> Result<Jwk, Error> {
+    pub fn new(rsa: Option<Rsa<Private>>, ca: &CA) -> Result<Jwk, Error> {
         let rsa: Rsa<Private> = match rsa {
             Some(rsa) => rsa,
             None => Rsa::generate(2048)?,
         };
 
         let modulus: Vec<u8> = rsa.n().to_vec();
         let exponent: Vec<u8> = rsa.e().to_vec();
-        let pkey: PKey<Private> = PKey::from_rsa(rsa)?;
+        let key_pair: PKey<Private> = PKey::from_rsa(rsa)?;
+
+        let x509 = ca.sign(&key_pair)?;
```

Comment:
Naive question: why don't we directly create a self-signed certificate using this key pair (instead of creating a ca to sign a certificate with this key)?

Reply:
Because the official sdk expects that the certificate is signed by a certification authority